indico/modules/rb_new/client/js/components/TimelineLegend.jsx

```diff
@@ -0,0 +1,60 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Label, Segment, List} from 'semantic-ui-react';
+import {Translate} from 'indico/react/i18n';
+
+import {legendLabelShape} from '../props';
+import './TimelineLegend.module.scss';
+
+
+export default function TimelineLegend({labels, aside, compact}) {
+    return compact ? (
+        <List styleName="legend compact">
+            {labels.map(({label, color, style}) => (
+                <List.Item key={label}>
+                    <List.Content styleName="labels">
+                        <Label color={color} styleName={`compact ${style || ''}`} />
+                        <span styleName="text">{Translate.string(label)}</span>
+                    </List.Content>
```

Comment:
I guess this `Translate` is not needed then?
Thanks for taking care of the PRs :wink:

Reply:
That would actually break during extraction since the value to translation functions cannot be dynamic. thanks for pointing it out, didn't notice it

Reply:
Fixed. Actually found a bug in the i18n lib while doing so ;)

Tip: `indico i18n extract_messages_react` spots such problems since we fail with an error in case of invalid translation logic calls such as passing a dynamic value to `Translate.string()`. We don't keep the pot file updated for now, but running the command and then reverting the change to the pot file afterwards is still useful to check that everything is valid.